unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarLOD/AvatarsLODController.cs

```diff
@@ -73,47 +76,68 @@ internal void UpdateLODsBillboard()
-                float distanceToPlayer = Vector3.Distance(CommonScriptableObjects.playerUnityPosition.Get(), position);
-                bool isInLODDistance = distanceToPlayer >= DataStore.i.avatarsLOD.LODDistance.Get();
-
-                if (isInLODDistance)
+                if (sortedPlayers[i].distance < 0) //Behind camera
                 {
-                    lodController.SetImpostorState();
+                    sortedPlayers[i].lodController.SetInvisible();
+                    continue;
                 }
-                else
+
+                if (sortedPlayers[i].distance < DataStore.i.avatarsLOD.LODDistance.Get())
                 {
-                    while (closeDistanceAvatars.ContainsKey(distanceToPlayer))
+                    if (avatarsCount >= maxAvatarVisible)
                     {
-                        distanceToPlayer += 0.0001f;
+                        sortedPlayers[i].lodController.SetInvisible();
+                        continue;
                     }
-                    closeDistanceAvatars.Add(distanceToPlayer, lodController);
+                    if (sortedPlayers[i].distance < SIMPLE_AVATAR_DISTANCE)
+                        sortedPlayers[i].lodController.SetAvatarState();
+                    else
+                        sortedPlayers[i].lodController.SetSimpleAvatar();
+
+                    avatarsCount++;
+                    continue;
                 }
-            }
 
-            int closeDistanceAvatarsCount = closeDistanceAvatars.Count;
-            for (var i = 0; i < closeDistanceAvatarsCount; i++)
-            {
-                IAvatarLODController currentAvatar = closeDistanceAvatars.Values[i];
-                bool isLOD = i >= DataStore.i.avatarsLOD.maxNonLODAvatars.Get();
-                if (isLOD)
-                    currentAvatar.SetImpostorState();
-                else
+                if (avatarsCount + impostorCount < maxPlayerVisible)
                 {
-                    if (closeDistanceAvatars.Keys[i] < SIMPLE_AVATAR_DISTANCE)
-                        currentAvatar.SetAvatarState();
-                    else
-                        currentAvatar.SetSimpleAvatar();
+                    sortedPlayers[i].lodController.SetImpostorState();
+                    impostorCount++;
+                    continue;
                 }
+
+                sortedPlayers[i].lodController.SetInvisible();
             }
         }
 
+        /// <summary>
+        /// Returns -1 if player is not in front of camera or not found
+        /// </summary>
+        /// <param name="player"></param>
+        /// <returns></returns>
+        private float DistanceToOwnPlayer(string playerId)
+        {
+            if (!otherPlayers.TryGetValue(playerId, out Player player))
+                return -1;
+
+            if (!IsInFrontOfCamera(player))
+                return -1;
+            return Vector3.Distance(CommonScriptableObjects.playerUnityPosition.Get(), player.worldPosition);
```

Comment:
Could we use a sqr distance for this if it is going to be called multiple times? :)

Reply:
We actually talked about that and in the chaos of t his PR I forgot to implement it, gonna do it!